### ISSUE TITLE
HADES Audit: Remove non-paper components and clarify architecture

### ISSUE THREAD

**Issue by toddwbucy** [toddwbucy/Atlas-MAG_OmegaRule#36]
## Summary

HADES-verified audit of Atlas paper (arXiv:2505.23735) identified components in our implementation that are **NOT in the paper** and should be removed for faithful implementation.

## Components to REMOVE (NOT in paper)

| Component | Current State | Action |
|-----------|---------------|--------|
| **Polarization loss** | Disabled via `--no-polarization` | ✅ Already handled |
| **Memory dropout** | Set to 0.0 | ✅ Already handled |
| **Per-layer gate init** | Gates init differently per layer (0.15→0.40) | ❌ Need to remove |
| **Blending gate** | Learnable `memory_gate` per block | ❓ Architectural decision needed |

## Hyperparameter to TUNE (in paper but not specified)

- **M_persistent initialization**: Currently `0.1 × I` (identity scaled by 0.1)
- Paper mentions persistent memory but doesn't specify initialization value
- **Action**: Keep as tunable hyperparameter for ablation studies

## Architectural Decision: Blending Gate

Our implementation combines attention AND memory in every block with a learned gate:
```python
x = x + attn_out + gate * mem_out  # Current implementation
```

The Atlas paper describes:
- **Pure Atlas**: Memory-only (no attention)
- **MAG/MAL hybrids**: INTERLEAVED layers (some SWA, some memory)

**Options:**

### Option A: Simple Addition (recommended for faithful impl)
```python
x = x + attn_out + mem_out  # Equal weight, no learned blend
```

### Option B: Match Paper's MAG - Interleaved Layers
```python
# Alternate: Layer 0 = SWA, Layer 1 = Memory, Layer 2 = SWA, etc.
```

### Option C: Keep Gate, Uniform Init
```python
# All gates start at 0.5, removes per-layer differentiation
```

## HADES Evidence

Key findings from paper queries:

1. **Gamma (γ) in paper** = decay weights INSIDE Omega Rule for token weighting
2. **No mention** of output-level blending gate between attention and memory
3. **MAG = "Memory After Gate"** refers to layer arrangement, not per-position gating
4. Paper quote: "we use input-dependent parameters for γ(t)_i, providing in-context pruning ability" - this is about context window decay, not attention/memory blend

## Proposed Action Plan

1. Remove per-layer gate initialization (all gates same init)
2. Decide on blending gate approach (Option A recommended)
3. Run ablation tests on M_persistent init values: `[0.01, 0.1, 0.5, 1.0]`
4. Document results and update implementation

## Related

- Current training: `atlas_faithful_v1` (polarization disabled, dropout 0.0)
- Audit report: `/home/todd/olympus/Atlas-MAG-audit/reports/CODEBASE_AUDIT.md`
